src/Functions/FunctionsJSON.cpp

```diff
@@ -1100,8 +1129,13 @@ struct JSONExtractTree
             case TypeIndex::LowCardinality:
             {
                 auto dictionary_type = typeid_cast<const DataTypeLowCardinality *>(type.get())->getDictionaryType();
+                if ((*dictionary_type).getTypeId() == TypeIndex::FixedString)
+                {
+                    auto fixed_length = typeid_cast<const DataTypeFixedString *>(dictionary_type.get())->getN();
+                    return std::make_unique<LowCardinalityFixedStringNode>(fixed_length);
+                }
                 auto impl = build(function_name, dictionary_type);
-                return std::make_unique<LowCardinalityNode>(dictionary_type, std::move(impl));
```

Comment:
How about LowCardinality(String)?

Reply:
Low cardinality for String and Numeric are processed in their respective previous functions in order to simplify code: `JSONExtractStringImpl` and `JSONExtractNumericImpl`
The problem was that for FixedString I needed to access the result from `getN()` as the fixed string length, which is why has a different call.

Reply:
Please, add some comments about general logic: how we handle LowCardinality. Also comment why we need this specialization for FixedString. Explanations in comments to PRs are hard to find later and this code looks hard to understand without piece of reasoning behind all this optimizations.

Reply:
Added, let me know if it's clear